src/components/DateInput.jsx

```diff
@@ -121,7 +121,8 @@ export default class DateInput extends React.Component {
           placeholder={placeholder}
           autoComplete="off"
           maxLength={10}
-          disabled={disabled || this.isTouchDevice}
+          disabled={disabled}
+          readonly={this.isTouchDevice}
```

Comment:
I think this needs to be camel case - https://facebook.github.io/react/docs/dom-elements.html#all-supported-html-attributes